addons/stock/models/stock_move.py

```diff
@@ -151,46 +148,116 @@ def _default_group_id(self):
+                    move.write({'pack_operation_ids': [(4, move_line.id)]})
+            elif len(move.pack_operation_ids) == 1:
+                move.pack_operation_ids[0].qty_done = move.quantity_done
+            else:
+                raise UserError("Cannot set the done quantity from this stock move, work directly with the move lines.")
+
     def _set_product_qty(self):
         """ The meaning of product_qty field changed lately and is now a functional field computing the quantity
         in the default product UoM. This code has been added to raise an error if a write is made given a value
         for `product_qty`, where the same write should set the `product_uom_qty` field instead, in order to
         detect errors. """
         raise UserError(_('The requested operation cannot be processed because of a programming error setting the `product_qty` field instead of the `product_uom_qty`.'))
 
-    @api.one
-    @api.depends('linked_move_operation_ids.qty')
-    def _get_remaining_qty(self):
-        self.remaining_qty = float_round(self.product_qty - sum(self.mapped('linked_move_operation_ids').mapped('qty')), precision_rounding=self.product_id.uom_id.rounding)
-
-    @api.one
-    @api.depends('state', 'quant_ids.lot_id', 'reserved_quant_ids.lot_id')
-    def _compute_lot_ids(self):
-        if self.state == 'done':
-            self.lot_ids = self.mapped('quant_ids').mapped('lot_id').ids
-        else:
-            self.lot_ids = self.mapped('reserved_quant_ids').mapped('lot_id').ids
 
     @api.one
-    @api.depends('reserved_quant_ids.qty')
+    @api.depends('pack_operation_ids.product_qty')
     def _compute_reserved_availability(self):
-        self.reserved_availability = sum(self.mapped('reserved_quant_ids').mapped('qty'))
+        """ Fill the `availability` field on a stock move, which is the actual reserved quantity
+        and is represented by the aggregated `product_qty` on the linked move lines. If the move
+        is force assigned, the value will be 0.
+        """
+        self.reserved_availability = sum(self.pack_operation_ids.mapped('product_qty'))
 
     @api.one
     @api.depends('state', 'product_id', 'product_qty', 'location_id')
     def _compute_product_availability(self):
+        """ Fill the `availability` field on a stock move, which is the quantity to potentially
+        reserve. When the move is done, `availability` is set to the quantity the move did actually
+        move.
+        """
         if self.state == 'done':
             self.availability = self.product_qty
         else:
-            quants = self.env['stock.quant'].search([('location_id', 'child_of', self.location_id.id), ('product_id', '=', self.product_id.id), ('reservation_id', '=', False)])
-            self.availability = min(self.product_qty, sum(quants.mapped('qty')))
+            if self.id:
```

Comment:
not needed anymore, before we made stock quant only work with sql queries and this was an issue